spec/nodeattr_interface_spec.rb

```diff
@@ -27,6 +27,23 @@
 require 'exceptions'
 
 RSpec.describe Metalware::NodeattrInterface do
+  context 'with setup1 genders' do
+    before do
+      SpecUtils.use_mock_genders(self, genders_file: 'setup1/genders')
+    end
+
+    describe '#nodes_in_primary_group' do
+      it 'returns only the nodes in the primary group' do
+        nodes = Metalware::NodeattrInterface.nodes_in_primary_group('group1')
+        nodes_not_in_primary_group = nodes.select do |node|
+          node.match(/\AnodeA\d\d\Z/).nil?
+        end
+        expect(nodes.length).to eq(10)
+        expect(nodes_not_in_primary_group.length).to eq(0)
```

Comment:
Yep - but the other part of my original comment was that I can't see why we need so many nodes in the group. The tests would be as good, if not better as they would be clearer, with just 2 or 3 nodes in each group, unless there's some reason I've missed why we need more.